### ISSUE TITLE
I want to implement stage1 in chatGPT. What should I do？

### ISSUE THREAD

**Issue by 1a2cjitenfei** [hpcaitech/ColossalAI#3119]
### 📚 The doc issue

How to implement supervised finetune in stage1？
ColossalAl Can it be realized？
What should I do specifically？

**Comment by Camille7777** [comment]
Hiii, thanks for your questions!

1. The model of stage 1 is actually an initial model, which can either be a pretrained model or SFT (supervised fine-tuning) model. The instructGPT paper indicates that they used the SFT model. Therefore, you could use a prompt dataset with label to fine-tune a GPT model with supervised learning in this stage. This model later will be used in sampling dataset for stage 2 by providing model outputs to prompts. For more implementation ideas, here is the [guideline of instructGPT](https://arxiv.org/abs/2203.02155) section 3 and C.1.

2. ColossalAI has provided GPT Parallel Training to reduce GPU resources and accelerate your training, you could refer this example to start your journel on using ColossalAI [Train GPT with Colossal-AI](https://github.com/hpcaitech/ColossalAI/tree/main/examples/language/gpt).

3. If you want to go through the whole RLHF training in a quick view, it's betther to use pretrained model in stage 1 (since the initial model is not necessarily need to be SFT) and start Stage 2 & 3 with the ColossalAI example [here](https://github.com/hpcaitech/ColossalAI/tree/main/applications/ChatGPT). Otherwise, you can follow our GPT tutorial mentioned above to realize stage 1 or wait for our simple stage 1 example which is under preparation :)

**Comment by binmakeswell** [comment]
This issue was closed due to inactivity. Thanks.
